src/dmd/dsymbolsem.d

```diff
@@ -4764,6 +4764,29 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
         {
             auto s = (*cldec.members)[i];
             s.dsymbolSemantic(sc2);
+
+            if (cldec.classKind == ClassKind.cpp)
+            {
+                const DtorDeclaration dtor = s.isDtorDeclaration();
+                if (dtor && !dtor.isFinal())
+                {
+                    assert(cldec.cppDtorVtblIndex == -1); // how does a class declare 2 destructors?
```

Comment:
This code can get executed again if semantic was deferred due to forward references (search for `addDeferredSemantic`). Instead of an assert, you could use the state `cppDtorVtblIndex` to avoid duplicate vtbl entries.